### DISCUSSION TITLE
"--major-only" option for "outdated" command

### DISCUSSION THREAD

**Discussion by halloei** [composer/composer#10438] [General]
I've written a CI job that runs every night and checks my PHP projects for available Composer package updates. The number of those updates are then displayed in a badge, so you can see at a glance if you should update your dependencies.

Currently the command is as follows: 
```shell
composer outdated --locked --no-dev --minor-only --format json  | jq --raw-output '.locked | length'
```
It...
* excludes dev dependencies
* includes only updates that can be done without hassle (as they won't break anything)
* outputs in JSON format so I can count the number of updates with `jq`

The output of this command is an integer which can be used to create the badge.

However, this won't include major updates from direct dependencies. I'd like to notice if there's a major update of one of them, because I can become active and update it. I thought about a second command with `--direct --major-only` and adding the result to the one of my previous command. But something like `--major-only` does not exist.

Would such an option be useful? How do you handle that kind of feature?

**Comment by Seldaek** [comment]
Created an issue for it https://github.com/composer/composer/issues/10439 IMO it'd be valuable
